sdk/core/src/http_client.rs

```diff
@@ -0,0 +1,22 @@
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct HttpClientArc(Arc<reqwest::Client>);
```

Comment:
I'm not sure if we gain too much from this new type. What's the thought behind a new type instead of using `Arc<Client>` directly?

Reply:
I like it. It has some value. It gives us `HttpClientArc::default()`, which should still work when we change the internals to be HttpClient interface. I think it also communicates how the library should be used.